_posts/2018-08-23-make-your-own-cli-with-golang-and-cobra.md

```diff
@@ -0,0 +1,224 @@
+---
+layout: post
+authors: [nick_geudens]
+title: 'Make your own CLI with golang and cobra'
+image: /img/make-your-own-cli-with-golang-and-cobra/banner.jpg
+tags: [Golang,Cobra,CLI,Go]
+category: Development
+comments: true
+---
+
+# Table of contents
+1. [Getting started](#getting-started)
+2. [Adding functionality](#adding-functionality)
+3. [Adding flags](#adding-flags)
+4. [Using environment variables instead of flags](#using-environment-variables-instead-of-flags)
+5. [Taking a file as input](#taking-a-file-as-input)
+6. [That’s it!](#thats-it)
+
+>There are lots of command line interfaces (CLI's) available these days, some of which are written in golang.
+There’s even a big chance that you’re using one of them.
+Docker or Kubernetes for example, have a CLI written in go.
+These were written using a framework called cobra.
+And that's exactly what I'm going to use to show you how to write your own simple CLI with golang.
+I’m also going to show you ways to provide parameters like environment variables and config files to your CLI, so you can do more that just the basics.
+
+# Getting started
+The framework Cobra provides a generator that adds some boilerplate code for you.
+This is handy because now you can focus more on the logic of your CLI instead of figuring out how to parse a flags.
+Assuming you have golang correctly installed, you can get the generator by doing the following.
+```bash
+go get github.com/spf13/cobra/cobra
+```
+This creates an executable which you can run from anywhere, because it is located in the $GOPATH/bin directory, which is in turn added to your PATH variable if you installed go correctly.
```

Comment:
Code styling can make things easier to read.
```suggestion
This creates an executable which you can run from anywhere, because it is located in the `$GOPATH/bin` directory, which is in turn added to your `PATH` variable if you installed Go correctly.
```